### ISSUE TITLE
Add documentation of the api of record object

### ISSUE THREAD

**Issue by 199911** [Airtable/airtable.js#32]
There is no documentation about the method in the object return by `select()`,
like  `updateFields()` and  `replaceFields()`. 
Please add it

**Comment by Vadorequest** [comment]
This article details the API more than what's in the README here: https://flaviocopes.com/airtable/

It's very nice that we have a dedicated and real time documentation for our own API, but having a better documentation here would also be nice!

**Comment by EvanHahn** [comment]
You're right, Airtable.js's documentation could be more thorough. We'll work on improving documentation as we iterate on this library.

You almost certainly know about this, but make sure to click the "Node" tab on the docs at [airtable.com/api](http://airtable.com/api)—it should help give you some guidance.

**Comment by Vadorequest** [comment]
Yup, indeed I noticed that "Node" tab and it's very handy to be honest, great feature!

It's just weird to go to an API library and not see any usage examples, it's not a big issue if you own an Airtable account but if you don't you have to create one and that's not good on the DX side ;)

**Comment by dylburger** [comment]
Hi @EvanHahn , really appreciate the library — the programmatically-generated docs for a given base are pretty cool.

I think I have a related question to this issue, so adding here. When I try to use the Node example from those API docs for retrieving details of a given record using the `find()` method, e.g.

```javascript
var Airtable = require('airtable');
var base = new Airtable({apiKey: 'myApiKey'}).base(
  'myBaseId',
);

base('Table 1').find(recordId), function(err, record) {
  if (err) {
    console.error(err);
    return;
  }
  console.log(record);
});
```

The output of `console.log(record)` looks something like this:

```
Class {
  _table:
   Class {
     _base: Class { _airtable: [Object], _id: 'appYDNu1RHfCtCujm' },
     id: null,
     name: 'Table 1',
     find: [Function],
     select: [Function: bound _selectRecords],
     create: [Function],
     update: [Function],
     replace: [Function],
     destroy: [Function],
     list: [Function],
     forEach: [Function] },
  id: 'recTW5iszw5ossqz9',
  _rawJson:
   { id: 'recTW5iszw5ossqz9',
     fields: { Notes: 'Blah', Attachments: [Array], Name: 'Test' },
     createdTime: '2019-06-18T21:58:09.000Z' },
  fields: { Notes: 'Blah', Attachments: [ [Object] ], Name: 'Test' },
  save: [Function],
  patchUpdate: [Function],
  putUpdate: [Function],
  destroy: [Function],
  fetch: [Function],
  updateFields: [Function],
  replaceFields: [Function] }
```

This output looks different than what the Node docs specify, so I just wanted to confirm I'm using the correct method, and using the callback in the correct way. I also tried using the promisified version of this method, e.g. using `async/await`:

```javascript
const record = await base(tableId).find(recordId)
```

but got the same object back (which makes sense, again just wanted to confirm).

Any advice would be appreciated. Thanks!

**Comment by EvanHahn** [comment]
@dylburger You're right, our docs are incorrect. I'll make sure to fix those in our API documentation—thanks for reporting!

**Comment by dylburger** [comment]
@EvanHahn for sure, really appreciate the response.

I'd follow up with a bit more feedback: this API response appears to include private properties (e.g. `_rawJson`) and other things that it seems like a consumer like me shouldn't rely on for a public-facing API. I'd expect a response closer to what the docs currently return — for example, I'd expect the `find` method to return something like the `fields` property, with the rest of the `_table`, `save`, and other methods removed.

Is that consistent with what y'all envision for the API? Is there a specific reason these properties were included in the response for the `find` method?

**Comment by Vadorequest** [comment]
Just reacting on the `_rawJson` thing, as I use it myself and so far it's perfectly fine. I believe it was added as "_" because it's not meant to be the most common use case, but it's actually quite useful to have it available under that format.

**Comment by EvanHahn** [comment]
It is a JavaScript convention to prefix private members with an underscore, like `_rawJson`. These should be considered library internals and may be removed in the future, so you shouldn't rely on them.
